src/server.c

```diff
@@ -1565,6 +1581,9 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         return;
     }
 
+    /* Pause clients on write as long as OOM and pending lazy-free jobs in background */
+    ClientWritePauseDuringOOM();
```

Comment:
isn't the one in processCommand enough? why do we need one in beforeSleep?

Reply:
This logic is also related to `performEvictions()` logic. 
We want to have rather updated state before reaching eviction phase.
I could have remove it but I preferred to be on the safe side and add
another checkpoint.